doc/docs/NLopt_Algorithms.md

```diff
@@ -123,6 +123,24 @@ Some references on StoGO are:
 
 Only bound-constrained problems are supported by this algorithm.
 
+### AGS
+
+This algorithm adapted from [this repo](https://github.com/sovrasov/glob_search_nlp_solver).
+AGS can handle arbitrary objectives and nonlinear inequality constraints. Also bound constraints are required for this method. To guarantee convergence, objectives and constraints should satisfy the Lipschitz condition on the specified hyperrectangle.
+AGS is derivative-free and employs the Hilbert curve to reduce the source problem to the univariate one. The algorithm divides the univariate space into intervals, generating new points by using posterior probabilities. On each trial AGS tries to evaluate the constraints consequently one by one. If some constraint is violated at this point, the next ones won't be evaluated. If all constraints are preserved, i.e. the trial point is feasible, AGS will evaluate the objective. Thus, some of constraints (except the first one) and objective can be partially undefined inside the search hyperrectangle. Current implementation of AGS doesn't support vector constraints.
+
+Limitations of the machine arithmetic don't allow to build a tight approximation for Hilbert when the space dimension is greater than 5, so this implementation of AGS is restricted in that sense.
```

Comment:
If it is just slightly less reliable for dimensions > 5, it is probably worth allowing it to be used there so that people can experiment… or does the code not work at all in higher dimensions?

Reply:
Ok, increased the limit to 10, added a warning that sometimes the method can stop early for high-dimensional problems.